source/content/diff/undoing_changes.md

```diff
@@ -0,0 +1,253 @@
+{% if slide %}
+### Going Back in Time: Undoing Changes
+{% else %}
+## Going Back in Time: Undoing Changes
+{% endif %}
+
+{% if page %}
+When you need to **undo changes** or **go back to a previous state**, Git offers several approaches. **Choose carefully** - some methods rewrite history!
+
+### Safe Methods (Preserve History)
+
+::::{grid}
+:::{grid-item-card} <i class="fa-solid fa-undo"></i> `git revert`
+:class-title: text-success
+```bash
+git revert <commit-hash>
+```
+**Creates a new commit** that undoes changes from a specific commit. **Safe for shared repositories**.
+:::
+:::{grid-item-card} <i class="fa-solid fa-archive"></i> `git stash`
+:class-title: text-success
+```bash
+git stash
+git stash pop
+```
+**Temporarily saves** uncommitted changes. **Safe** - doesn't modify history.
+:::
+::::
+
+### Destructive Methods (Rewrite History)
+
+```{danger}
+**Warning**: These commands rewrite history. **Never use on shared/pushed commits** unless you coordinate with your team!
+```
+
+::::{grid}
+:::{grid-item-card} <i class="fa-solid fa-eraser"></i> `git reset --soft`
+:class-title: text-warning
+```bash
+git reset --soft HEAD~1
+```
+**Moves HEAD** back but keeps changes staged. Safest reset option.
+:::
+:::{grid-item-card} <i class="fa-solid fa-step-backward"></i> `git reset --mixed`
+:class-title: text-warning
+```bash
+git reset HEAD~1
+# or 
+git reset --mixed HEAD~1
+```
+**Moves HEAD** back and unstages changes, but keeps them in working directory.
+:::
+::::
+
+::::{grid}
+:::{grid-item-card} <i class="fa-solid fa-trash"></i> `git reset --hard`
+:class-title: text-danger
+```bash
+git reset --hard HEAD~1
+```
+**⚠️ DESTRUCTIVE**: Moves HEAD back and **deletes all changes**. Use with extreme caution!
+:::
+:::{grid-item-card} <i class="fa-solid fa-code-branch"></i> `git rebase`
+:class-title: text-warning
+```bash
+git rebase -i HEAD~3
+```
+**Rewrites commit history**. Powerful but **dangerous** on shared branches.
+:::
+::::
+
+### When to Use Which Method?
+
+:::::{dropdown} **Decision Guide: Which "go back" method to use?**
+:color: primary
+
+**Q: Are the commits already pushed/shared with others?**
+
+- **YES** → Use `git revert` (creates new commit, safe for shared history)
+- **NO** → You can use destructive methods, but consider the consequences
+
+**Q: Do you want to keep your changes?**
+
+- **Keep as staged changes** → `git reset --soft`
+- **Keep as unstaged changes** → `git reset --mixed` (default)
+- **Discard all changes** → `git reset --hard` ⚠️
+
+**Q: Do you want to fix the history/commit messages?**
+
+- **YES** → `git rebase -i` (interactive rebase)
+- **NO** → Use `git reset` methods
+
+**Q: Do you just want to temporarily save work?**
```

Comment:
Instead of _work_ we should say un-committed changes (i.e. edits - staged or not - to tracked files).

Reply:
in fact, if the history are sequences of commits then `git stage` does not actually work on the history at all, as it only handles un-commited changes.